packages/transformer/src/IModelTransformer.ts

```diff
@@ -2222,9 +2222,14 @@ export class IModelTransformer extends IModelExportHandler {
         i++
       )
         syncChangesetsToUpdate.push(i);
-      // FIXME: add test to synchronize an iModel that is not at the tip, since then clearning syncChangesets is
-      // probably wrong, and we should filter it instead
-      syncChangesetsToClear.length = 0;
+      // Only keep the changeset indices which are greater than the source, this means they haven't been processed yet.
+      syncChangesetsToClear.splice(
+        0,
+        syncChangesetsToClear.length,
+        ...syncChangesetsToClear.filter((csIndex) => {
+          return csIndex > this._startingChangesetIndices!.source;
+        })
+      );
```

Comment:
nit: this is very confusing for a delete. Could we either stick with the `.length` idiom but calculate the length like:
```ts
syncChangesetsToClear.length = syncChangesetsToClear.findIndex(c => c > this._startingChangesetIndices.source);
```
or save the key to edit (`syncChangesetsToClearKey`) and replace the list rather than mutate it?
```ts
jsonProps[syncChangesetsToClearKey] = syncChangesetsToClear.filter();
```